### PR TITLE
fixed #614 multiple captions of the same language all appeared as selected when selecting only one

### LINKED ISSUE #614: [Bug]: Multiple subtitles of the same language appear both selected
### What happened?

image

### What browsers are you seeing the problem on?

Other (tell us in input box below)

### Steps to reproduce?

Browser: Vivaldi

Bug visible on [Full House S01E01](https://movie-web.app/media/tmdb-tv-4313-full-house/12718/297397)

### Other relevant information

_No response_

### PR DESCRIPTION
**Pull request by CelestialLemon**
This pull request resolves #614

 - [x] I have read and agreed to the [code of conduct](https://github.com/movie-web/movie-web/blob/dev/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/movie-web/movie-web/blob/dev/.github/CONTRIBUTING.md).
 - [x] What I'm implementing was assigned to me and is an [approved issue](https://github.com/movie-web/movie-web/issues?q=is%3Aopen+is%3Aissue+label%3Aapproved). For reference, please take a look at our [GitHub projects](https://github.com/movie-web/movie-web/projects).
 - [x] I have tested all of my changes.

IDs for the captions were added to the providers package recently. Now instead of using language to set the selected caption we can use the ID of that captions. This fixes the issue where we have multiple subtitles of the same language which all appeared selected when selecting only one of them. 

It's my first PR to this project and I'm new here so let me know if anything else is required to be done here. Thanks.

### CODE REVIEW

File: src/components/player/atoms/settings/CaptionsView.tsx [hunk 1]
```diff
@@ -140,14 +141,14 @@ export function CaptionsView({ id }: { id: string }) {
         // eslint-disable-next-line react/no-array-index-key
         key={`${i}-${v.url}`}
```
Comment (mrjvs) [inline src/components/player/atoms/settings/CaptionsView.tsx:142]: I think this key can also be with the id. no longer needs the eslint disable comment too

File: src/components/player/hooks/useCaptions.ts [hunk 2]
```diff
@@ -14,12 +14,30 @@ export function useCaptions() {
...
   const selectLanguage = useCallback(
     async (language: string) => {
       const caption = captionList.find((v) => v.language === language);
       if (!caption) return;
       const srtData = await downloadCaption(caption);
       setCaption({
+        id: caption.id,
         language: caption.language,
         srtData,
         url: caption.url,
```
Comment (mrjvs) [inline src/components/player/hooks/useCaptions.ts:43]: This method should call the new `selectCaptionById` after its found the correct caption.

**Review by mrjvs — CHANGES_REQUESTED** [review]
Look good overal, just two small bits of cleanup and then it should be good to go.

**Comment by CelestialLemon** [comment]
Changes done.

**Comment by CelestialLemon** [comment]
Done.

**Review by mrjvs — APPROVED** [review]

**Pull request merged**